File: backend-java/src/main/java/com/app/service/SentimentService.java
import org.springframework.web.client.RestTemplate;
import org.springframework.http.ResponseEntity;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpHeaders;
import org.springframework.stereotype.Service;
import java.util.HashMap;
import java.util.Map;

@Service
public class SentimentService {
    private static final String API_URL = "http://localhost:5000/analyze";

    public String analyzeSentiment(String text) {
        RestTemplate restTemplate = new RestTemplate();
        HttpHeaders headers = new HttpHeaders();
        headers.set("Content-Type", "application/json");

        Map<String, String> request = new HashMap<>();
        request.put("text", text);

        HttpEntity<Map<String, String>> entity = new HttpEntity<>(request, headers);
        ResponseEntity<Map> response = restTemplate.postForEntity(API_URL, entity, Map.class);

        return response.getBody().get("sentiment").toString();
    }
}

File: frontend/src/services/sentimentService.js
export const analyzeSentiment = async (text) => {
    try {
        const response = await fetch("http://localhost:8080/api/sentiment", {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify({ text }),
        });
        const data = await response.json();
        return data.sentiment;
    } catch (error) {
        console.error("Error:", error);
        return "Error";
    }
};
